core/src/main/java/com/valtech/aem/saas/core/indexing/DefaultIndexUpdateService.java

```diff
@@ -0,0 +1,154 @@
+            .setContentEncoding(StandardCharsets.UTF_8.name())
+            .build())
+        .build();
+
+    return searchRequestExecutorService.execute(searchRequest)
+        .flatMap(response -> response.get(new DefaultIndexUpdateDataExtractionStrategy()));
+  }
+
+  @Override
+  public Optional<IndexUpdateResponse> deleteIndexedUrl(@NonNull String client, @NonNull String url,
+      @NonNull String repositoryPath) {
+    validateInput(client, url, repositoryPath);
+    SearchRequest searchRequest = SearchRequestDelete.builder()
+        .uri(getRequestUri(client, configuration.indexUpdateService_apiIndexTriggerAction()))
+        .httpEntity(EntityBuilder.create().setParameters(new BasicNameValuePair(REQUEST_PARAMETER_URL, url),
+                new BasicNameValuePair(REQUEST_PARAMETER_REPOSITORY_PATH, repositoryPath))
+            .setContentEncoding(StandardCharsets.UTF_8.name())
+            .build())
+        .build();
+    return searchRequestExecutorService.execute(searchRequest)
+        .flatMap(response -> response.get(new DefaultIndexUpdateDataExtractionStrategy()));
+  }
+
+  @Override
+  public Optional<IndexUpdateResponse> indexContent(@NonNull String client,
+      @NonNull IndexContentPayload indexContentPayload) {
+    validateInputClient(client);
+    SearchRequest searchRequest = SearchRequestPost.builder()
+        .uri(getRequestUri(client, configuration.indexUpdateService_apiPushContentAction()))
+        .httpEntity(EntityBuilder.create().setText(new Gson().toJson(indexContentPayload))
+            .setContentType(ContentType.APPLICATION_JSON)
+            .setContentEncoding(StandardCharsets.UTF_8.name())
+            .build())
+        .build();
+
+    return searchRequestExecutorService.execute(searchRequest)
+        .flatMap(response -> response.get(new DefaultIndexUpdateDataExtractionStrategy()));
+  }
+
+  private String getRequestUri(String client, String action) {
+    return String.format("%s%s%s%s%s", searchServiceConnectionConfigurationService.getBaseUrl(),
+        configuration.indexUpdateService_apiBasePath(), client,
+        configuration.indexUpdateService_apiVersionPath(), action);
+  }
+
+  private void validateInput(String client, String url, String repositoryPath) {
+    validateInputClient(client);
+    if (StringUtils.isBlank(url)) {
+      throw new IllegalArgumentException("Please pass a url of a content that should be indexed.");
+    }
+    if (StringUtils.isBlank(repositoryPath)) {
+      throw new IllegalArgumentException("Please pass a repository path regex value.");
+    }
+  }
+
+  private void validateInputClient(String client) {
+    if (StringUtils.isBlank(client)) {
+      throw new IllegalArgumentException("Please configure client in context aware configuration.");
+    }
+  }
+
+  @ObjectClassDefinition(name = "Search as a Service - Index Update Service Configuration",
+      description = "Index Update Api specific details.")
+  public @interface Configuration {
```

Comment:
this service is huge, what do you think, shall we extract this Configuration in separate file for better readability?

Reply:
Imo since it is  designated specifically to this osgi component, i'd say it is ok if it stays in the same file.